cvat-core/src/cloud-storage.js

```diff
@@ -0,0 +1,407 @@
+                                }
+                            }
+                        },
+                    },
+                    /**
+                     * Session token
+                     * @name token
+                     * @type {string}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     * @throws {module:API.cvat.exceptions.ArgumentError}
+                     */
+                    token: {
+                        get: () => data.token,
+                        set: (value) => {
+                            if (typeof value === 'string') {
+                                if (value.trim().length) {
+                                    data.token = value;
+                                } else {
+                                    throw new ArgumentError('Value must not be empty');
+                                }
+                            }
+                        },
+                    },
+                    /**
+                     * Unique resource name
+                     * @name resourceName
+                     * @type {string}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     * @throws {module:API.cvat.exceptions.ArgumentError}
+                     */
+                    resourceName: {
+                        get: () => data.resourceName,
+                        set: (value) => {
+                            if (!value.trim().length) {
+                                throw new ArgumentError('Value must not be empty');
+                            }
+                            data.resourceName = value;
+                        },
+                    },
+                    /**
+                     * @name manifestPath
+                     * @type {string}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     */
+                    manifestPath: {
+                        get: () => data.manifestPath,
+                        set: (value) => {
+                            if (typeof value !== undefined) {
+                                data.manifestPath = value;
+                            }
+                        },
+                    },
+                    /**
+                     * @name provider
+                     * @type {module:API.cvat.enums.ProviderType}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     * @throws {module:API.cvat.exceptions.ArgumentError}
+                     */
+                    provider: {
+                        get: () => data.provider,
+                        set: (value) => {
+                            if (value !== undefined && !!ProviderType[value]) {
+                                data.provider = value;
```

Comment:
As I understand from this code ``value`` is not a value, is a key.
Should not we write ``data.provider = ProviderType[value]``? (and actually rename ``value`` to ``key``?